tests/integer_date_attribute_test.py

```diff
@@ -28,14 +30,15 @@ def test_serialization_non_null(uuid_key):
     model.save()
 
     # verify underlying storage
-    item = MyModel._get_connection().get_item(uuid_key)
+    item = _connection(MyModel).get_item(uuid_key)
     assert item['Item'] == {'key': ANY, 'value': {'N': '20151231'}}
 
     # verify deserialization
     model = MyModel.get(uuid_key)
-    assert model.value.year == 2015
-    assert model.value.month == 12
-    assert model.value.day == 31
+    value = cast(date, model.value)
```

Comment:
Without this:

```console
$ pre-commit run mypy --all-files
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

tests/integer_date_attribute_test.py:39: error: "IntegerDateAttribute" has no attribute "year"
tests/integer_date_attribute_test.py:40: error: "IntegerDateAttribute" has no attribute "month"
tests/integer_date_attribute_test.py:41: error: "IntegerDateAttribute" has no attribute "day"
```

Reply:
I believe we wouldn't need this once you add pynamodb 5.0.1b to pre-commit mypy `additional_dependencies`.

Reply:
Is that one released?

```
stderr:
    ERROR: Could not find a version that satisfies the requirement pynamodb==5.0.1b (from versions: 0.1.5, 0.1.6, 0.1.7, 0.1.8, 0.1.9, 0.1.10, 0.1.11, 0.1.12, 0.1.13, 1.0.0, 1.1.0, 1.2.0, 1.2.1, 1.2.2, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.3.8, 1.4.0, 1.4.2, 1.4.3, 1.4.4, 1.5.0, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.6.0, 2.0.0, 2.0.1, 2.0.2, 2.0.3, 2.1.4, 2.1.5, 2.1.6, 2.2.0, 3.0.1, 3.1.0, 3.2.0rc1, 3.2.0rc2, 3.2.0, 3.2.1, 3.3.0, 3.3.1, 3.3.2, 3.3.3, 3.4.0, 3.4.1, 4.0.0a1, 4.0.0b1, 4.0.0b2, 4.0.0b3, 4.0.0, 4.1.0, 4.2.0, 4.3.0, 4.3.1, 4.3.2)
    ERROR: No matching distribution found for pynamodb==5.0.1b
```

Reply:
Bah, I thought we released a pre-release. Might then add `pynamodb==4.3.2`.

Reply:
The issue persists

```diff
diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index 6cade9a..6770220 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -46,3 +46,5 @@ repos:
       rev: v0.710
       hooks:
         - id: mypy
+          additional_dependencies:
+            - pynamodb==4.3.2
diff --git a/tests/integer_date_attribute_test.py b/tests/integer_date_attribute_test.py
index 91580d3..c2082c8 100644
--- a/tests/integer_date_attribute_test.py
+++ b/tests/integer_date_attribute_test.py
@@ -1,5 +1,4 @@
 from datetime import date
-from typing import cast
 from unittest.mock import ANY
 
 import pytest
@@ -35,10 +34,9 @@ def test_serialization_non_null(uuid_key):
 
     # verify deserialization
     model = MyModel.get(uuid_key)
-    value = cast(date, model.value)
-    assert value.year == 2015
-    assert value.month == 12
-    assert value.day == 31
+    assert model.value.year == 2015
+    assert model.value.month == 12
+    assert model.value.day == 31
 
 
 def test_serialization_null(uuid_key):
```

```console
$ pre-commit run mypy --all-files
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

tests/integer_date_attribute_test.py:37: error: "IntegerDateAttribute" has no attribute "year"
tests/integer_date_attribute_test.py:38: error: "IntegerDateAttribute" has no attribute "month"
tests/integer_date_attribute_test.py:39: error: "IntegerDateAttribute" has no attribute "day"
```

Reply:
I removed the cast and it still seems to think it's a `date`

Reply:
Well I guess we'll see what CI says because the issue persists in my repro. Do you have a different repro/setup?

Reply:
CI is red